components/network/CircumventionToolsChart.js

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo } from 'react'
+import { useRouter } from 'next/router'
+import { Heading, Box, Flex } from 'ooni-components'
+import axios from 'axios'
+import { axiosResponseTime } from 'components/axios-plugins'
+import useSWR from 'swr'
+import GridChart, { prepareDataForGridChart } from 'components/aggregation/mat/GridChart'
+import { DetailsBox } from 'components/measurement/DetailsBox'
+
+const swrOptions = {
+  revalidateOnFocus: false,
+  dedupingInterval: 10 * 60 * 1000,
+}
+
+const baseURL = process.env.NEXT_PUBLIC_AGGREGATION_API
+axiosResponseTime(axios)
+
+const circumventionFetcher = (query) => {
+  const reqUrl = `${baseURL}/api/v1/aggregation?${query}`
+  const result = []
+
+  const request = (reqUrl) => { 
+    return axios.get(reqUrl).then(r => {
+      if (!r?.data?.result) {
+        const error = new Error(`Request ${reqUrl} did not contain expected result`)
+        error.data = r
+        throw error
+      }
+      return {
+        data: r.data.result,
+        loadTime: r.loadTime,
+        url: r.config.url
+      }
+    }).catch(e => {
+      e.message = e?.request?.response ?? e.message
+      throw e
+    })
+  }
```

Comment:
I see this is used in both the IM app page and this one, perhaps this can be refactored into some shared function?